### PR TITLE
Issue 55 create status tracker

### PR DESCRIPTION
**Pull request by DhanashriPalodkar**
**What type of PR is this?**

kind feature
This PR creates resources for status tracker. Generates Yaml file

**What does does this PR do / why we need it**:

**Which issue(s) this PR fixes**:
Gitops- issue-55 
https://issues.redhat.com/browse/GITOPS-55

### CODE REVIEW

File: pkg/pipelines/roles/service_rolebinding_test.go [hunk 1]
```diff
@@ -94,20 +107,9 @@ func TestCreateRole(t *testing.T) {
...
-				APIGroups: []string{"tekton.dev"},
-				Resources: []string{"pipelineruns", "pipelineresources", "taskruns"},
-				Verbs:     []string{"create"},
-			},
-		},
+		Rules: testRules,
 	}
-	roleTask := createRole(meta.NamespacedName("", roleName), rules)
+	roleTask := CreateRole(meta.NamespacedName("", roleName), testRules)
 	if diff := cmp.Diff(roleTask, want); diff != "" {
```
Comment (wtam2018) [inline pkg/pipelines/roles/service_rolebinding_test.go:104]: I think the origin intent of this test including the validation of the "roles" (as defined in bootstrap.go).   Should we keep that?

File: pkg/pipelines/statustracker/deployment.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/odo/pkg/pipelines/meta"
+	"github.com/openshift/odo/pkg/pipelines/roles"
+)
+
+const (
+	operatorName   = "commit-status-tracker"
+	containerImage = "quay.io/kmcdermo/status-tracker"
+)
```
Comment (wtam2018) [inline pkg/pipelines/statustracker/deployment.go:21]: Got 403 Unauthorized when I tried tdo pull this image.   Please make sure it is readable to everyone.

File: pkg/pipelines/statustracker/deployment.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+			Verbs:     []string{"get"},
+		},
+
+		rbacv1.PolicyRule{
+			APIGroups: []string{"tekton.dev"},
+			Resources: []string{"pipelineruns"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+	}
+)
```
Comment (wtam2018) [inline pkg/pipelines/statustracker/deployment.go:70]: TypeMeta is missing. The output yaml will fail to apply because of missing "Kind".

Comment (DhanashriPalodkar) [reply pkg/pipelines/statustracker/deployment.go:70]: sure William I will look into it

File: pkg/pipelines/roles/service_rolebinding_test.go (hunk 1 above)
Comment (bigkevmcd) [reply pkg/pipelines/roles/service_rolebinding_test.go:104]: I don't think that this is that valuable.

Unless you are taking a clean-box approach to the test, i.e. making sure that there's no copy-paste of the rules from one to the other, you are not really testing the rules.

This test ensures that if you give it a set of rules, those are put into a Role.

Validating the rules is a tricky proposition.

File: pkg/pipelines/statustracker/deployment.go (hunk 2 above)
Comment (bigkevmcd) [reply pkg/pipelines/statustracker/deployment.go:21]: It is, but we should probably put it under the quay.io/redhat-developer organisation.

Comment (bigkevmcd) [reply pkg/pipelines/statustracker/deployment.go:21]: I would, but I can't create new repositories in the redhat-developer org, @sbose78 can you help?

Comment (sbose78) [reply pkg/pipelines/statustracker/deployment.go:21]: Made `kmcdermo` an owner in redhat-developer 👍 .

File: pkg/pipelines/roles/service_rolebinding_test.go (hunk 1 above)
Comment (wtam2018) [reply pkg/pipelines/roles/service_rolebinding_test.go:104]: As pointed out, this change removed the validation of roles defined bootstrap.go.   This change makes the test even less valuable.

File: pkg/pipelines/statustracker/deployment.go (hunk 2 above)
Comment (wtam2018) [reply pkg/pipelines/statustracker/deployment.go:21]: @bigkevmcd  Please provide the source repo of the image.

Comment (bigkevmcd) [reply pkg/pipelines/statustracker/deployment.go:21]: https://github.com/tektoncd/experimental/tree/master/commit-status-tracker

File: pkg/pipelines/bootstrap.go [hunk 4]
```diff
@@ -140,18 +142,19 @@ func Bootstrap(o *BootstrapParameters) error {
 
 	//  Create Role, Role Bindings, and ClusterRole Bindings
 	outputs = append(outputs, createRoleBindings(namespaces, sa)...)
+	outputs = append(outputs, statustracker.Resources(namespaces["cicd"])...)
```
Comment (wtam2018) [inline pkg/pipelines/bootstrap.go]: Status tracker resource should not be added if status-tracker-token is not provided.    

Also, the status-tracker-token option sets up the "github-auth" secret.   What is "github-auth" consumed by status tracker?  We should rename the field "GitHubToken" to something like "StatusTrackerToken" to reflect their relationship.

**Pull request merged**